File: leetcode/1. Two Sum/TwoSum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> m;
        vector<int> result;
        for (int i = 0; i < nums.size(); i++)
        {
            if (m.find(target - nums[i]) != m.end())
            {
                result.push_back(i);
                result.push_back(m.find(target - nums[i])->second);
                return result;
            }
            else
            {
                m.emplace(nums[i], i);
            }
        }
        return result;
    }
};

File: leetcode/1. Two Sum/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const m = new Map();
    for (let i in nums) {
        if (m.has(target - nums[i])) {
            return [i, m.get(target - nums[i])];
        } else {
            m.set(nums[i], i);
        }
    }
    return -1;
};